app/src/main/java/chat/rocket/android/inviteusers/presentation/InviteUsersPresenter.kt

```diff
@@ -0,0 +1,85 @@
+package chat.rocket.android.inviteusers.presentation
+
+import chat.rocket.android.core.lifecycle.CancelStrategy
+import chat.rocket.android.db.DatabaseManager
+import chat.rocket.android.members.uimodel.MemberUiModelMapper
+import chat.rocket.android.server.infrastructure.RocketChatClientFactory
+import chat.rocket.android.util.extension.launchUI
+import chat.rocket.common.RocketChatException
+import chat.rocket.common.model.roomTypeOf
+import chat.rocket.common.util.ifNull
+import chat.rocket.core.RocketChatClient
+import chat.rocket.core.internal.rest.invite
+import chat.rocket.core.internal.rest.spotlight
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.withContext
+import javax.inject.Inject
+import javax.inject.Named
+
+class InviteUsersPresenter @Inject constructor(
+	private val view: InviteUsersView,
+	private val dbManager: DatabaseManager,
+	@Named("currentServer") private val currentServer: String,
+	private val strategy: CancelStrategy,
+	private val mapper: MemberUiModelMapper,
+	val factory: RocketChatClientFactory
+) {
+	private val client: RocketChatClient = factory.get(currentServer)
+
+	fun inviteUsers(chatRoomId: String, usersList: List<Pair<String, String>>) {
+		launchUI(strategy) {
+			view.disableUserInput()
+			view.showLoading()
+
+			val stringBuilder = StringBuilder()
+
+			try {
+				for (user in usersList) {
+					try {
```

Comment:
Why are you using `try { ...` again here?

Reply:
For each user it makes API call to server. And for each user we need to catch exception like `user already added in the room` or `something went wrong`

Outer try block is used to for hiding loading and showing msg after API call is made for all the user in the list.